_project/api/_src/Usecases/Blog.Controllers.ts

```diff
@@ -0,0 +1,76 @@
+import { BlogPost } from "@effect-app-boilerplate/models/Blog"
+import { BlogRsc } from "@effect-app-boilerplate/resources"
+import { BogusEvent } from "@effect-app-boilerplate/resources/Events"
+import { PositiveInt } from "@effect-app/prelude/schema"
+import { BlogPostRepo, Events, Operations } from "api/services.js"
+
+const blog = matchFor(BlogRsc)
+
+const FindPost = blog.matchFindPost(
+  { BlogPostRepo },
+  (req, { blogPostRepo }) =>
+    blogPostRepo
+      .find(req.id)
+      .map((_) => _.getOrNull)
+)
+const GetPosts = blog.matchGetPosts(
+  { BlogPostRepo },
+  (_, { blogPostRepo }) => blogPostRepo.all.map((items) => ({ items }))
+)
+
+const CreatePost = blog.matchCreatePost(
+  { BlogPostRepo },
+  (req, { blogPostRepo }) =>
+    Effect(new BlogPost({ ...req }))
+      .tap(blogPostRepo.save)
+      .map((_) => _.id)
+)
+
+const PublishPost = blog.matchPublishPost(
+  { BlogPostRepo, Events, Operations },
```

Comment:
One of the main reasons to switch to fp/effect from imperative was going point-free and nested codeblocks, can't we simplify this piece? 🙈

Reply:
I've played a long time with the apis, so far this is very convenient :) and also familiar.
familiar is good, you still reap all the FP benefits.

Reply:
You can happily use pipe baed apis though, but I find it harder to sway people to FP with piped based apis